packages/playwright/src/runner/testRunner.ts

```diff
@@ -331,7 +331,7 @@ export class TestRunner extends EventEmitter<TestRunnerEventMap> {
 
     config.cliListOnly = false;
     config.cliPassWithNoTests = true;
-    config.cliArgs = params.locations || [];
+    config.cliArgs = params.locations ?? ['dont-match-anything']; // for performance reasons, we require test runner users to pass `locations`. they can opt out by passing [].
```

Comment:
If we require passing locations, let's make them required in the type, instead of using `dont-match-anything` hack.